src/wasm2js.h

```diff
@@ -448,9 +448,9 @@ static std::string base64Encode(std::vector<char> &data) {
 
   while (i + 3 <= data.size()) {
     int bits =
-      (((int) data[i + 0]) << 16) |
-      (((int) data[i + 1]) << 8) |
-      (((int) data[i + 2]) << 0);
+      (((uint8_t) data[i + 0]) << 16) |
```

Comment:
Hmm, wait, this doesn't seem right - as `uint8_t` it would just be 0 after doing `<< 16`, wouldn't it? I think we want `uint32_t` here.

Please also make sure we have a test that verifies that.

Reply:
the attached test verifies just that.

Reply:
the `(uint8_t)` cast is needed to fix the bug. the "the integer promotions" extends it to integer. I can be explicit and add `(uint32_t)` everywhere.

Reply:
Great, thanks, I think it's less surprising to read that way. lgtm.